esda/tests/test_correlogram.py

```diff
@@ -0,0 +1,22 @@
+from libpysal import examples
+from esda import correlogram
+import numpy as np
+from numpy.testing import assert_array_almost_equal
+
+import geopandas as gpd
+
+
+sac = gpd.read_file(examples.load_example("Sacramento1").get_path("sacramentot2.shp"))
+sac = sac.to_crs(sac.estimate_utm_crs())  # now in meters)
+
+distances = [i + 500 for i in range(0, 2000, 500)]
+
+def test_correlogram():
```

Comment:
Do we have some ecosystem-wide standards on what is the minimal acceptable test suite?

Reply:
No... We've talked about separating "correctness" tests (does the statistic recover the same result every time?) and option tests/user tests (does the function handle all combination of arguments/input types correctly?), but we've never agreed on what is necessary for any contribution...

Reply:
Yup, makes sense. I asked because I am not sure when should I be fine with the state of tests when doing the review, so an agreed guideline would be nice.

Reply:
Very much agree with Martin on this. Seems like our general rule of thumb has been that somewhere between 70-90% is the acceptable coverage. I always try to get as close to 100% as possible to (attempt to) hit any weird edge cases, but I know that isn't required a lot of the time. If we want to decide on an actual number, this is surely something for at least the steering committee to vote on. Moreover, (1) as a step towards catching edge cases we may want to consider seeing about implementing [`hypothesis`](https://hypothesis.readthedocs.io/en/latest/), and (2) we should improve testing across all submodules to properly test minimum requirements, [as Martin points out here](https://github.com/pysal/esda/pull/258).

Reply:
>If we want to decide on an actual number, this is surely something for at least the steering committee to vote on. Moreover, (1) as a step towards catching edge cases we may want to consider seeing about implementing [hypothesis](https://hypothesis.readthedocs.io/en/latest/), and (2) we should improve testing across all submodules to properly test minimum requirements, https://github.com/pysal/esda/pull/258.

again, i dont think an ecosystem-wide policy is realistic, but definitely something we should discuss. The question is about cost-benefit. If we're too strict about test coverage, it will discourage contributions, and i think for many packages, that's a greater risk than having some untested code. I don't have time to get absorbed in edge-cases, so I'd prefer to let those issues surface before letting them block new functionality getting merged.

case in point, please feel free to write out the rest of the test suite here :)

Reply:
> > If we want to decide on an actual number, this is surely something for at least the steering committee to vote on. Moreover, (1) as a step towards catching edge cases we may want to consider seeing about implementing [hypothesis](https://hypothesis.readthedocs.io/en/latest/), and (2) we should improve testing across all submodules to properly test minimum requirements, #258.
> 
> again, i dont think an ecosystem-wide policy is realistic, but definitely something we should discuss. The question is about cost-benefit. If we're too strict about test coverage, it will discourage contributions, and i think for many packages, that's a greater risk than having some untested code. I don't have time to get absorbed in edge-cases, so I'd prefer to let those issues surface before letting them block new functionality getting merged.
> 
> case in point, please feel free to write out the rest of the test suite here :)

FWIW @knaaptime I am "on your side" here (though I think we're all on the same side). In my opinion, getting new functionality merged should take precedence over a strict coverage number for individual PRs, so long as that new functionality is being "mostly" covered (where "mostly" can either be more lines covered *or* majority cases covered). More testing can be added later as time & energy permit. I hope my comments did not come off as combative.

Reply:
😅 sorry my style is always blunt! no offense taken or intended. We're all on the same page. We need some guidelines on merging--definitely. I just want to avoid a situation where we let stuff sit here, because it's usually easier for us core maintainers to take incremental passes at improving test coverage, instead of imposing a big burden on the PR opener

(for this one, I certainly don't *need* this PR merged, but while it's sitting here it makes for a good example. I already have the function myself, but if other folks want to use it, i'd prefer not to let corner cases prevent that, since i know i don't have a ton of time to write out the test suite. I can usually commit to responding to bug reports though)

Reply:
My question was primarily coming from my GeoPandas experience and understanding that what we aim for there is not applicable in PySAL. So I was just wondering what is the level people feel is enough. Totally agree with all you said above, there's no need to block stuff just because tests are not 100%. Happy to have a further chat about it during the next call. And happy to merge this with the test suite as is :).

Reply:
Once again I agree with Martin, but maybe after addressing https://github.com/pysal/esda/pull/259#discussion_r1284119611?